src/test/java/org/folio/impl/EventConfigAPIsTest.java

```diff
@@ -0,0 +1,197 @@
+      .extract()
+      .response();
+    actualEntity = new JsonObject(responseGetEvent.getBody().print());
+    assertEquals(expectedEntity, actualEntity);
+
+    // delete the event config by id
+    requestDeleteEntityById(id)
+      .then()
+      .statusCode(HttpStatus.SC_OK);
+
+    // check that the event config has been deleted
+    requestGetEventById(id)
+      .then()
+      .statusCode(HttpStatus.SC_BAD_REQUEST);
+  }
+
+  @Test
+  public void testPostEventConfig() {
+    String id = UUID.randomUUID().toString();
+    JsonObject expectedEntity = getJsonEntity(id, "test name", false, new JsonArray());
+
+    Response response = requestPostEventConfig(expectedEntity)
+      .then()
+      .statusCode(HttpStatus.SC_OK)
+      .extract()
+      .response();
+
+    JsonObject actualEntity = new JsonObject(response.getBody().print());
+    assertEquals(expectedEntity, actualEntity);
+
+    requestDeleteEntityById(id)
+      .then()
+      .statusCode(HttpStatus.SC_OK);
+  }
+
+  @Test
+  public void testGetEventConfigById() {
+    String id = UUID.randomUUID().toString();
+    requestGetEventById(id)
+      .then()
+      .statusCode(HttpStatus.SC_BAD_REQUEST);
+  }
+
+  @Test
+  public void testPutEventConfigById() {
+    String id = UUID.randomUUID().toString();
+    JsonObject entity = getJsonEntity("0002", "test", true, new JsonArray());
+
+    requestPutEventConfig(id, entity)
+      .then()
+      .statusCode(HttpStatus.SC_BAD_REQUEST);
+  }
+
+  @Test
+  public void testDeleteEventConfigById() {
+    String id = UUID.randomUUID().toString();
+    requestDeleteEntityById(id)
+      .then()
+      .statusCode(HttpStatus.SC_BAD_REQUEST);
+  }
+
+  private JsonObject getJsonEntity(String id, String name, boolean active, JsonArray templates) {
+    return new JsonObject()
+      .put("id", id)
+      .put("name", name)
+      .put("active", active)
+      .put("templates", templates);
+  }
+
+  private Response requestPostEventConfig(JsonObject expectedEntity) {
+    return RestAssured.given()
+      .port(port)
+      .contentType(MediaType.APPLICATION_JSON)
+      .headers(headers)
+      .body(expectedEntity.toString())
+      .when()
+      .post(REST_PATH);
+  }
+
+  private Response requestGetEventById(String expectedId) {
+    return RestAssured.given()
+      .port(port)
+      .contentType(MediaType.APPLICATION_JSON)
+      .headers(headers)
+      .when()
+      .get(String.format(PATH_TEMPLATE, REST_PATH, expectedId));
+  }
+
+  private Response requestPutEventConfig(String id, JsonObject entity) {
+    return RestAssured.given().port(port)
+      .contentType(MediaType.APPLICATION_JSON)
+      .headers(headers)
+      .body(entity.toString())
+      .when()
+      .put(String.format(PATH_TEMPLATE, REST_PATH, id));
+  }
+
+  private Response requestDeleteEntityById(String id) {
+    return RestAssured.given()
+      .port(port)
+      .contentType(MediaType.APPLICATION_JSON)
+      .headers(headers)
```

Comment:
You can put this repeatable stuff into RequestSpecification and use it
RestAssured.given()
      .spec(spec)
      .when() ... your code here

Reply:
thanks for the comments.
made changes to the code.